packages/drawer/src/Drawer/Drawer.tsx

```diff
@@ -85,9 +88,12 @@ export const Drawer = forwardRef<HTMLDivElement, DrawerProps>(
       if (open) {
         registerDrawer(id);
       } else {
-        setTimeout(() => unregisterDrawer(id), drawerTransitionDuration);
+        setTimeout(
+          () => unregisterDrawer(id),
+          prefersReducedMotion ? 0 : drawerTransitionDuration,
+        );
       }
-    }, [id, open, registerDrawer, unregisterDrawer]);
+    }, [id, open, prefersReducedMotion, registerDrawer, unregisterDrawer]);
```

Comment:
Not part of this PR, but wouldn't this result in multiple registered drawers with the same id if the component re-renders?